app/api/api_v1/routers/search.py

```diff
@@ -203,22 +174,45 @@ def search_documents(
 def download_search_documents(
     request: Request,
     search_body: SearchRequestBody,
+    app_token: Annotated[str, Header()],
     db=Depends(get_db),
 ) -> StreamingResponse:
     """Download a CSV containing details of documents matching the search criteria."""
+    token = AppTokenFactory()
 
     _LOGGER.info(
         "Search download request",
         extra={
             "props": {
                 "search_request": search_body.model_dump(),
+                "app_token": str(app_token),
             }
         },
     )
+
+    # Decode the app token and validate it.
+    #
+    # First corpora validation is app token against DB. At least one of the app token
+    # corpora IDs must be present in the DB to continue the search request.
+    token = AppTokenFactory()
```

Comment:
COMMENT: You may well end up writing some middleware to do this

Reply:
@diversemix what do you mean here exactly?

Reply:
Yeah - totally agree!